### PR TITLE
test(web-carousel): add tests for `web-carousel`, fixes #7191

### LINKED ISSUE #7191: Write tests for `web-carousel`
Write tests for `web-carousel`

### PR DESCRIPTION
**Pull request by MichaelSolati**
Fixes #7191

Changes proposed in this pull request:

- Update rollup plugins, required for `testConfig`/`istanbul` to handle optional chaining
- Update Carousel to set `index` as a public property
- Add tests for Carousel
- Build and include `next.scss` in `test:lib` command

When you're ready to submit your PR, don't forget to add the `$-presubmit` label.

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -46,6 +46,13 @@ const virtualImports = {
 const pagesDir = './src/lib/pages/';
 const pages = fs.readdirSync(pagesDir, 'utf-8').map((p) => join(pagesDir, p));
 
+// Suppress circular dependency warning
```
Comment (devnook) [inline rollup.config.js:49]: Why do we need to suppress it?

Comment (MichaelSolati) [reply rollup.config.js:49]: We don't need to, but it's a bit of an annoying message.

File: test/unit/src/lib/components/Carousel/index.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const {expect, assert} = require('chai');
+require('../../../../../../src/lib/components/Carousel/index');
+
+const cardCount = 10;
+const sleep = () => new Promise((res) => setTimeout(res, 600));
```
Comment (devnook) [inline test/unit/src/lib/components/Carousel/index.js:22]: Please add comment why it needs to wait 600 ms

File: test/unit/src/lib/components/Carousel/index.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+ * @returns {Promise<import('../../../../../../src/lib/components/Carousel/index').Carousel>}
+ */
+const setup = async () => {
+  const webCarousel =
+    /** @type {import('../../../../../../src/lib/components/Carousel/index').Carousel} */ (
+      document.createElement('web-carousel')
+    );
+
+  const divCarousel = document.createElement('div');
+  divCarousel.classList.add('carousel');
```
Comment (devnook) [inline test/unit/src/lib/components/Carousel/index.js:34]: You can try to use jsdom next time instead of creating complicated html manually htis way.

**Review by devnook — APPROVED** [review]
Left some nits

**Pull request merged**